warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/UniqueByDay.java

```diff
@@ -0,0 +1,56 @@
+package datawave.query.language.functions.jexl;
+
+import datawave.query.jexl.functions.QueryFunctions;
+import datawave.query.language.functions.QueryFunction;
+import datawave.webservice.query.exception.BadRequestQueryException;
+import datawave.webservice.query.exception.DatawaveErrorCode;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+
+/**
+ * Function to return a unique result for every day for a given list of fields. This function is equivalent to {@code #unique(field[DAY])}.
+ */
+public class UniqueByDay extends JexlQueryFunction {
+    
+    public UniqueByDay() {
+        super(QueryFunctions.UNIQUE_BY_DAY_FUNCTION, new ArrayList<>());
+    }
+    
+    /**
+     * query options contain a list of fields. Cannot be the empty list.
+     * 
+     * @throws IllegalArgumentException
+     */
+    @Override
+    public void validate() throws IllegalArgumentException {
+        if (this.parameterList.isEmpty()) {
+            BadRequestQueryException qe = new BadRequestQueryException(DatawaveErrorCode.INVALID_FUNCTION_ARGUMENTS, MessageFormat.format("{0}", this.name));
+            throw new IllegalArgumentException(qe);
+        }
+    }
+    
+    @Override
+    public String toString() {
```

Comment:
Can this function be moved a super class (e.g. UniqueByFunction)? It appears the same in the other UniqueBy... classes.

Reply:
Done.